### PR TITLE
Prevent nil-pointer exception in Validate()

### PR DESCRIPTION
**Pull request by binacs**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

```go
func (o *SomeOptions) Validate() []error {
	// Some code logic...
	o.SomeSructMemberA = someValue	// `nil` panic!
	o.SomeSructMemberB.SomeFunc()	// `nil` panic!
	// ...
}
```

#### Which issue(s) this PR fixes:

Prevent nil-pointer exception in `Validate()` just like this:

```go
func (o *SomeOptions) Validate() []error {
	if o == nil {
		return nil
	}

	// Some code logic...
	o.SomeSructMemberA = someValue
	o.SomeSructMemberB.SomeFunc()
	// ...
}
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by libaoan — APPROVED** [review]
done

**Comment by BinacsLee** [comment]
cc @ehashman @Huang-Wei 
Thanks

File: cmd/kube-scheduler/app/options/options.go [hunk 1]
```diff
@@ -232,6 +232,10 @@ func emptySchedulerProfileConfig(profiles []kubeschedulerconfig.KubeSchedulerPro
 
 // Validate validates all the required options.
 func (o *Options) Validate() []error {
+	if o == nil {
```
Comment (Huang-Wei) [inline cmd/kube-scheduler/app/options/options.go:235]: The only case `o` can be nil is:

https://github.com/kubernetes/kubernetes/blob/6d130d3b97242a6d9cfc2d9dc4d72030f441ec80/cmd/kube-scheduler/app/options/options.go#L75-L79

However, if `newDefaultComponentConfig` returns an error, it should have already been fatal at:

https://github.com/kubernetes/kubernetes/blob/ed3e0d302fb546653b78df583569b0311687a7a8/cmd/kube-scheduler/app/server.go#L64-L68

So this nil check looks unneeded.

Comment (BinacsLee) [reply cmd/kube-scheduler/app/options/options.go:235]: Thanks for review！

I've considered that some check are now unnecessary. But in fact, there are many such unnecessary checks in project. Such a check is always safer while we don't know when the code logic will change.

**Comment by BinacsLee** [comment]
Hi @logicalhan , the tests has been passed ^_^

**Comment by ehashman** [comment]
/close

Thank you for your PR, @BinacsLee. I've reviewed your PR and it seems to only contain trivial edits. Please read our [trivial edits policy](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#trivial-edits). We encourage you to take a look at confirmed issues and bugs or issues marked `help-wanted`.

**Pull request closed without merging**